modules/vstudio/vs2010_vcxproj_filters.lua

```diff
@@ -47,18 +47,55 @@
 
 --
 -- The first portion of the filters file assigns unique IDs to each
--- directory or virtual group. Would be cool if we could automatically
--- map vpaths like "**.h" to an <Extensions>h</Extensions> element.
+-- directory or virtual group.
 --
 
 	function m.uniqueIdentifiers(prj)
-		local tr = project.getsourcetree(prj)
+		local knownFilters = {
+			['Source Files'] = {
+				order = 1,
+				extensions = 'cpp;c;cc;cxx;def;odl;idl;hpj;bat;asm;asmx',
```

Comment:
My initial thought is that these extensions should match what we have in the generator for determining what's a source file for the vcxproj file. Where did these lists come from?

Reply:
These lists correspond to the contents of the generated files if you do "New > Project" in the IDE itself and so are, IMO, the "canonical" extensions for the corresponding kinds of files. So for me using the actual extensions used in vpaths would result in gratuitous (because they don't really affect anything) differences between Premake-generated and "native" project files and I'd like to avoid this because I want to review the diffs when replacing the existing build system with Premake.

AFAIK these extensions are mostly cosmetic anyhow, i.e. they affect the list of files shown when adding files to this group in the IDE, but this is never going to happen anyhow for the generated projects.

Reply:
From the perspective of Premake, I think it's more correct to make the filters match what's in the `ClCompile`, `ClInclude`, etc. lists in the `vcxproj` generator.  We'll end up with filter mismatches with the file extensions in the project file, which isn't something we want.

Reply:
OK, what about applying this PR without the last commit then? This would just change the order of the filters, without changing anything else.

Reply:
I'm fine with leaving out the last commit.

Reply:
Should I force push to this PR to remove the last commit or will you just merge `FETCH_HEAD~`?

AFAICS no new tests are needed as the existing tests cover this already (and had to be updated).

Reply:
Force pushed now.